### DISCUSSION TITLE
How to set a variable?

### DISCUSSION THREAD

**Discussion by avila** [cmderdev/cmder#2494] [Q&A] [answered]
Within `Git Bash` I can set and substitue a variable as following: 

```
x=1
echo $x
``` 

How can I do this in `cmder`?

(I have the strong feeling this is a stupid question, but I really couldn't figure it out)

**Comment by daxgames** [comment, marked as answer]
Cmder default session is based on  `cmd.exe`  which uses `*.bat` or` *.cmd` as scripts.  This is often referred to as DOS batch or simply batch.

```
set x=1
echo %x%
```

You can also use Git Bash inside Cmder by launching a `bash::bash` session.
